@shared/api/internal.ts

```diff
@@ -47,13 +47,23 @@ export const fundAccount = async (publicKey: string): Promise<void> => {
 
 export const addAccount = async (
   password: string,
-): Promise<{ publicKey: string; allAccounts: Array<Account> }> => {
+): Promise<{
+  publicKey: string;
+  allAccounts: Array<Account>;
+  hasPrivateKey: boolean;
+}> => {
   let error = "";
   let publicKey = "";
   let allAccounts = [] as Array<Account>;
+  let hasPrivateKey = false;
 
   try {
-    ({ allAccounts, error, publicKey } = await sendMessageToBackground({
+    ({
+      allAccounts,
+      error,
+      publicKey,
+      hasPrivateKey,
```

Comment:
the biggest change in this PR is starting the private key "timer" whenever a user enters their password.

So, for ex, if a user had to enter their password to import a new account, save the private key so we don't need to show the "unlock account" prompt for the next 24h

Reply:
`hasPrivateKey` is the trigger for showing "unlock account" on screens like Manage Assets and Send Payment

Reply:
I thought the `hasPrivateKey` was available to the popup from the[ `auth` field ](https://github.com/stellar/freighter/blob/018c263420c71f3bc2f1e64a2a1cb3f7247c47a3/extension/src/popup/ducks/accountServices.ts#L227)in the redux store. Why does it need to be returned from these methods as well? Is it because it doesn't update in the popup redux after calling these methods (ie addAccount, recoverAccount)?

Reply:
Yup, you got it. Previously, addAccount/recoverAccount/etc didn't start the timer. Now that they do, these methods will return a new `hasPrivateKey` value that will be used in the popup redux duck to update the field. For example: [extension/src/popup/ducks/accountServices.ts](https://github.com/stellar/freighter/pull/438/files#diff-3d5d228c5caf387ac8dbfe41239ed0a40dd86588386fb9674839819681053905)

Reply:
ah I see, cool thanks for explaining